x-pack/plugin/stack/src/main/java/org/elasticsearch/xpack/stack/StackTemplateRegistry.java

```diff
@@ -253,6 +258,42 @@ protected List<LifecyclePolicy> getLifecyclePolicies() {
 
     @Override
     protected Map<String, ComponentTemplate> getComponentTemplateConfigs() {
+        if (logsIndexModeEnabled) {
+            final IndexTemplateConfig override = new IndexTemplateConfig(
+                LOGS_SETTINGS_COMPONENT_TEMPLATE_NAME,
+                "/logs@settings-logsdb.json",
+                REGISTRY_VERSION,
+                TEMPLATE_VERSION_VARIABLE,
+                ADDITIONAL_TEMPLATE_VARIABLES
+            );
+            final Map<String, ComponentTemplate> updatedComponentTemplates = new HashMap<>(COMPONENT_TEMPLATE_CONFIGS);
```

Comment:
Had to do this because the original map is immutable and I need access to node settings.

Reply:
Maybe create two static immutable maps and return the right one based on the setting?